server/game/cards/07_LAW/units/TheSarlaccOfCarkoonHorrorOfTheDuneSea.ts

```diff
@@ -0,0 +1,40 @@
+import type { IAbilityHelper } from '../../../AbilityHelper';
+import type { INonLeaderUnitAbilityRegistrar } from '../../../core/card/AbilityRegistrationInterfaces';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, WildcardCardType, ZoneName } from '../../../core/Constants';
+
+export default class TheSarlaccOfCarkoonHorrorOfTheDuneSea extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: 'the-sarlacc-of-carkoon#horror-of-the-dune-sea-id',
+            internalName: 'the-sarlacc-of-carkoon#horror-of-the-dune-sea',
+        };
+    }
+
+    public override setupCardAbilities(registrar: INonLeaderUnitAbilityRegistrar, AbilityHelper: IAbilityHelper) {
+        registrar.addOnAttackAbility({
+            title: 'Put a unit from your discard pile on the bottom of your deck. Deal damage equal to that unit\'s power to an enemy ground unit',
+            targetResolvers: {
```

Comment:
I think there is 2 solutions for your problem here, maybe you could use an implicit then in the implementation or instead of using simultaneous on the second part, you should put the discarded card to the bottom of deck on the first part and deal damage on the second